build.zig

```diff
@@ -101,9 +101,23 @@ pub fn build(b: *std.Build) !void {
         }),
     });
     example.root_module.addImport("z3", z3_bindings);
-
+    // enable running `valgrind --leak-check=full zig-out/bin/main`
+    example.root_module.valgrind = true;
```

Comment:
Should `enable-valgrind` be a build option?  I'm actually not entirely sure what this does.  I assume it instruments the exe so that zig allocators can be detected/used by valgrind?  If thats true then I guess it should be removed since we're no longer using zig allocators.